src/app/space/create/deployments/deployments.component.ts

```diff
@@ -4,14 +4,25 @@ import { Spaces } from 'ngx-fabric8-wit';
 import { Observable, Subscription } from 'rxjs';
 
 import { Environment } from './models/environment';
-import { DeploymentsService } from './services/deployments.service';
+import {
+  DeploymentsService,
+  TIMER_TOKEN
+} from './services/deployments.service';
 
 @Component({
   encapsulation: ViewEncapsulation.None,
   selector: 'alm-apps',
   templateUrl: 'deployments.component.html',
   styleUrls: ['./deployments.component.less'],
-  providers: [DeploymentsService]
+  providers: [
+    DeploymentsService,
+    {
+      provide: TIMER_TOKEN,
```

Comment:
Why does this need to be provided outside of the DeploymentsService?  It seems a bit odd to inject a timer here instead of just creating it inside the deployments service - is there a best practice or some example you followed that you could link to explaining this possibly?

Reply:
The timer was just created inside the service previously, but I have a commit here making it injectable instead so that the unit test suite can inject a different implementation. Namely, a plain Subject so that the unit tests can easily control the update cycle of the service, rather than an actual timer. This could probably also have been done with fakeAsync, but replacing the timer altogether seemed like it would lead to more robust and maintainable tests.

Reply:
Ok - makes sense - not sure how this pattern will scale as we go, it's a little confusing to expose this as a shared observable but let's give it a shot.

Reply:
The other solution I had in mind was to make the observable protected and then have the tests subclass the service and override it that way, which also means the tests get to drop DI and just construct the service directly. That might be more clear to readers but was a more invasive change for the DeploymentsService, which is why I opted against it for this PR.

Alternatively the timer could just be a public assignable property and the tests simply reassign it, but that felt pretty wrong.